docs/tutorials/control/air-quality-fleet.md

```diff
@@ -0,0 +1,953 @@
+   You'll create a Viam `dataClient` instance which accesses all the data in your location, and then query this data to get only the data tagged with the `air-quality` tag you applied with your data service configuration.
+   The following code also queries the data for a list of the machines that have collected air quality data so that later, you can make a dashboard that has a place for the latest data from each of them.
+
+   Paste the following code into the main function of your <file>main.ts</file> script, directly after the `locationID` line, in place of `// <Insert data client and query code here in later steps>`:
+
+   ```typescript {class="line-numbers linkable-line-numbers"}
+   // Instantiate data_client and get all
+   // data tagged with "air-quality" from your location
+   const client = await VIAM.createViamClient(opts);
+   const myDataClient = client.dataClient;
+   const query = {
+     $match: {
+       tags: "air-quality",
+       location_id: locationID,
+       organization_id: orgID,
+     },
+   };
+   const match = { $group: { _id: "$robot_id" } };
+   // Get a list of all the IDs of machines that have collected air quality data
+   const BSONQueryForMachineIDList = [
+     BSON.serialize(query),
+     BSON.serialize(match),
+   ];
+   let machineIDs: any = await myDataClient?.tabularDataByMQL(
+     orgID,
+     BSONQueryForMachineIDList,
+   );
+   // Get all the air quality data
+   const BSONQueryForData = [BSON.serialize(query)];
+   let thedata: any = await myDataClient?.tabularDataByMQL(
+     orgID,
+     BSONQueryForData,
+   );
+   ```
+
+1. For this project, your dashboard will display the average of the last five readings from each air sensor.
+   You need a function to calculate that average.
+   The data returned by the query is not necessarily returned in order, so this function must put the data in order based on timestamps before averaging the last five readings.
+
+   Paste the following code into <file>main.ts</file> after the end of your main function, in place of `// <Insert getLastFewAv function definition here in later steps>`:
+
+   ```typescript {class="line-numbers linkable-line-numbers"}
+   // Get the average of the last few readings from a given sensor
+   async function getLastFewAv(alltheData: any[], machineID: string) {
+     // Get just the data from this machine
+     let thedata = new Array();
+     for (const entry of alltheData) {
+       if (entry.robot_id == machineID) {
+         thedata.push({
+           PM25: entry.data.readings["pm_2.5"],
+           time: entry.time_received,
+         });
+       }
+     }
+
+     // Sort the air quality data from this machine
+     // by timestamp
+     thedata = thedata.sort(function (a, b) {
+       let x = a.time.toString();
+       let y = b.time.toString();
+       if (x < y) {
+         return -1;
+       }
+       if (x > y) {
+         return 1;
+       }
+       return 0;
+     });
```

Comment:
Would this not accomplish the same thing? Presumably time is already a number? 
```suggestion
     thedata = thedata.sort((a, b) => a.time - b.time);
```
if not maybe
```
     thedata = thedata.sort((a, b) => Date.parse(a.time) - Date.parse(b.time));
```

Reply:
It's not a number to start with but I think your second suggestion works! Nice! Will apply to full code too

Reply:
```suggestion
     thedata = thedata.sort((a, b) => Date.parse(a.time) - Date.parse(b.time));
```